gradle-jdks-setup/src/main/java/com/palantir/gradle/jdks/setup/CaResources.java

```diff
@@ -53,113 +62,93 @@ public CaResources(ILogger logger) {
         this.logger = logger;
     }
 
-    public Optional<AliasContentCert> readPalantirRootCaFromSystemTruststore() {
-        return systemCertificates().flatMap(CaResources::selectPalantirCertificate);
+    public static Optional<AliasContentCert> readPalantirRootCaFromSystemTruststore() {
+        return selectPalantirCertificate(systemCertificates());
     }
 
-    public Optional<AliasContentCert> maybeGetCertificateFromSerialNumber(String serialNumber, String alias) {
-        return systemCertificates()
-                .map(certs -> selectCertificates(certs, Map.of(serialNumber, alias)))
-                .orElseGet(Stream::empty)
-                .findFirst();
+    public void importAllSystemCerts(Path jdkInstallationDirectory) {
+        importCertificates(jdkInstallationDirectory, parseCerts(systemCertificates()));
     }
 
-    public void importCertInJdk(AliasContentCert aliasContentCert, Path jdkInstallationDirectory) {
-        Os os = CurrentOs.get();
-        switch (os) {
-            case MACOS:
-            case LINUX_GLIBC:
-            case LINUX_MUSL:
-                unixImportCertInJdk(aliasContentCert, jdkInstallationDirectory);
-                logger.log(String.format(
-                        "Successfully imported CA certificate %s into the JDK truststore",
-                        aliasContentCert.getAlias()));
-                break;
-            case WINDOWS:
-                logger.logError(
-                        String.format("Importing certificates for OS type '%s' is not yet supported", os.uiName()));
-                break;
-        }
-    }
-
-    private void unixImportCertInJdk(AliasContentCert aliasContentCert, Path jdkInstallationDirectory) {
+    private void importCertificates(Path jdkInstallationDirectory, List<Certificate> certificates) {
         try {
-            if (isCertificateInTruststore(jdkInstallationDirectory, aliasContentCert.getAlias())) {
-                logger.log("Certificate already exists in the truststore, skipping...");
-                return;
+            char[] passwd = "changeit".toCharArray();
+            Path jksPath = jdkInstallationDirectory.resolve("lib/security/cacerts");
+            KeyStore jks = loadKeystore(passwd, jksPath);
+            Set<BigInteger> existingSerialNumbers = getExistingCertificates(jks);
```

Comment:
Serial number is actually bad here. I had a look in my system truststore, and there are 8 certs with serial number `0`, 6 with serial number `1` etc:

<img width="1530" alt="Screenshot 2024-09-06 at 14 10 12" src="https://github.com/user-attachments/assets/614ecea6-1f2c-41e0-97a3-5b1632c0582c">

Reply:
We probably want to use some hash of the certificate text somehow... there's `X509Certificate#getSignature` but it returns a byte[] - could probably turn that into a hex string?

Reply:
I think the `<CA, serialNumber>` pairs should be unique. I'll try to change the alias to account for this. 
I think the 0 serial number might be indeed invalid. Maybe a revoked certificate ? I will try to see if I can check the validity of the serialNumber somehow

Reply:
<img width="508" alt="Screenshot 2024-09-06 at 3 17 28 PM" src="https://github.com/user-attachments/assets/c4ed6e6e-c1f8-4a10-8dd9-5fab3490ef4f">